tests/functional/aws-node-sdk/test/object/putVersion.js

```diff
@@ -89,8 +89,13 @@ describe('PUT object with x-scal-s3-version-id header', () => {
                 next => s3.putObject(params, next),
                 next => fakeMetadataArchive(bucketName, objectName, undefined, archive, next),
                 next => getMetadata(bucketName, objectName, undefined, (err, objMD) => {
+                    if (err) {
+                        return next(err);
+                    }
+                    // eslint-disable-next-line no-param-reassign
+                    objMD.legalHold = true;
                     objMDBefore = objMD;
-                    return next(err);
+                    return metadata.putObjectMD(bucketName, objectName, objMD, params, log, err => next(err));
```

Comment:
should not overwrite metadata manually, this breaks the semantics of the scenario, no?